packages/p2p/src/stores/advertiser-page-store.js

```diff
@@ -107,13 +107,15 @@ export default class AdvertiserPageStore extends BaseStore {
                     ? { local_currency: buy_sell_store.selected_local_currency }
                     : {}),
             }).then(response => {
-                if (response.error) {
-                    this.setErrorMessage(response.error);
-                } else {
-                    const { list } = response.p2p_advert_list;
-
-                    this.setAdverts(list);
-                    this.setHasMoreAdvertsToLoad(list.length >= general_store.list_item_limit);
+                if (response) {
+                    if (response.error) {
+                        this.setErrorMessage(response.error);
+                    } else {
+                        const { list } = response.p2p_advert_list;
```

Comment:
```suggestion
                        const { list = [] } = response.p2p_advert_list;
```
Can the list (returned by BE) be undefined?

Reply:
just checked, the list wont be undefined, and there will always be an array returned *unless theres an error

Reply:
just that the above error couldn't recognise response.error, so we needed to check for response first